test/dynamo/test_activation_checkpointing.py

```diff
@@ -625,54 +693,69 @@ def fn(x, y):
+                    _get_custom_policy(no_recompute_list=no_recompute_list)
+                )
 
-        def gn(x, y):
-            return torch.sigmoid(
-                torch.matmul(torch.matmul(torch.bernoulli(torch.sigmoid(x)), y), y)
-            )
+            def gn(x):
+                return torch.sigmoid(torch.dropout(torch.sigmoid(x), p=0.5, train=True))
+
+            def fn(x):
+                return torch.utils.checkpoint.checkpoint(
+                    gn,
+                    x,
+                    use_reentrant=False,
+                    # Regardless of whether `preserve_rng_state` is True or False,
+                    # we will always preserve RNG state when using `torch.compile`.
+                    preserve_rng_state=preserve_rng_state,
+                    context_fn=selective_checkpointing_context_fn,
+                )
 
-        def fn(x, y):
-            return torch.utils.checkpoint.checkpoint(
-                gn,
-                torch.sin(x),
-                y,
-                use_reentrant=False,
-                context_fn=selective_checkpointing_context_fn,
-            )
+            x = torch.randn(4, 4, requires_grad=True, device="cuda")
 
-        x = torch.randn(4, 4, requires_grad=True)
-        y = torch.randn(4, 4, requires_grad=True)
+            fw_compiler = functools.partial(
+                count_ops,
+                freqs=[2, 1],
+                ops=[
+                    torch.ops.aten.sigmoid.default,
+                    torch.ops.aten.native_dropout.default,
+                ],
+            )
+            bw_compiler = functools.partial(
+                count_ops,
+                # NOTE: This unit test expects `dropout` to be recomputed (notice the count for `native_dropout` is 1).
+                freqs=[0, 1],
+                ops=[
+                    torch.ops.aten.sigmoid.default,
+                    torch.ops.aten.native_dropout.default,
+                ],
+            )
+            backend = aot_autograd(
+                fw_compiler=fw_compiler,
+                bw_compiler=bw_compiler,
+                partition_fn=min_cut_rematerialization_partition,
+            )
 
-        fw_compiler = functools.partial(
-            count_ops,
-            freqs=[2, 2],
-            ops=[torch.ops.aten.mm.default, torch.ops.aten.sigmoid.default],
-        )
-        bw_compiler = functools.partial(
-            count_ops,
-            freqs=[4, 0],
-            ops=[torch.ops.aten.mm.default, torch.ops.aten.sigmoid.default],
-        )
-        backend = aot_autograd(
-            fw_compiler=fw_compiler,
-            bw_compiler=bw_compiler,
-            partition_fn=min_cut_rematerialization_partition,
-        )
-        self._validate(fn, backend, x, y)
+            # NOTE: when `preserve_rng_state` is False, gradient will mismatch between torch.compile and eager,
+            # because eager version doesn't preserve RNG state while torch.compile still does.
```

Comment:
hmm I would expect if eager says "preserve_rng_state=False", then torch.compile should respect that?

Reply:
yeah I am a little unsure what's the use case for `preserve_rng_state=False` in eager and if we should support that in compile. For now I just mention that under compile, `preserve_rng_state` is always True. I believe we can change this approach based on user's feedback